src/Database/Type/FloatType.php

```diff
@@ -65,18 +65,18 @@ public function __construct($name = null)
      *
      * @param string|resource $value The value to convert.
      * @param \Cake\Database\Driver $driver The driver instance to convert with.
-     * @return string|resource
+     * @return string|null
      */
     public function toDatabase($value, Driver $driver)
     {
         if ($value === null || $value === '') {
             return null;
         }
         if (is_array($value)) {
-            return 1;
+            return '1';
         }
 
-        return (float)$value;
+        return sprintf('%F', (float)$value);
```

Comment:
Decimal is expected to work pretty much like Integer and both are expected to never be automatically transformed, such as truncated. Not having worked with neither BC Math or GMP is what Int and Decimal may require?
http://stackoverflow.com/questions/211345/working-with-large-numbers-in-php http://www.php.net/manual/en/book.bc.php
http://www.php.net/manual/en/book.gmp.php